src/admin/components/forms/field-types/Number/index.tsx

```diff
@@ -86,24 +87,32 @@ const NumberField: React.FC<Props> = (props) => {
         label={label}
         required={required}
       />
-      <input
-        id={`field-${path.replace(/\./gi, '__')}`}
-        value={typeof value === 'number' ? value : ''}
-        onChange={handleChange}
-        disabled={readOnly}
-        placeholder={getTranslation(placeholder, i18n)}
-        type="number"
-        name={path}
-        step={step}
-        onWheel={(e) => {
-          // eslint-disable-next-line @typescript-eslint/ban-ts-comment
-          // @ts-ignore
-          e.target.blur();
-        }}
-      />
+      <div className="number__wrap">
+        <input
+          id={`field-${path.replace(/\./gi, '__')}`}
+          value={typeof value === 'number' ? value : ''}
+          onChange={handleChange}
+          disabled={readOnly}
+          placeholder={getTranslation(placeholder, i18n)}
+          type="number"
+          name={path}
+          step={step}
+          onWheel={(e) => {
+            // eslint-disable-next-line @typescript-eslint/ban-ts-comment
+            // @ts-ignore
+            e.target.blur();
+          }}
+        />
+        {formatOptions && (
+          <div className="number__formatted">
+            {new Intl.NumberFormat(i18n.language, formatOptions).format(value as number)}
```

Comment:
This is great.

Reply:
Err, we need to format display `value` on line 93 actually—not render an additional div next to the input. I didn't catch this at first pass.

Reply:
This is not possible, the input type is set to `number`. For what you are asking, I believe we would need to set it to `type='tel'`, and to include some new javascript package that auto-formats the input as the user is typing. I disagree that we need to format inline on the input.

Reply:
The ultimate goal here is to format only the display value of the field without saving to the db, exactly as we currently do for dates. I see your point about the input being of `type="number"`. The solution here is to absolutely position a text field overtop the number field, and format the value of _that_. Then sync the value of the number field beneath, which is hidden.

Reply:
The date field relies on `react-datepicker`, I have attempted to use the `react-number-format` library but it doesn't handle decimal values with Intl.NumberFormat, not sure why.